alibi/explainers/tests/test_simiarlity/conftest.py

```diff
@@ -143,18 +144,26 @@ def linear_models(request):
     return tf_model, tf_loss, torch_model, torch_loss
 
 
-def tf_linear_model(input_shape, output_shape):
+def tf_linear_model(input_shape, output_shape, batch_norm=False):
     """
     Constructs a linear model for `tensorflow`.
     """
-    return keras.Sequential([
-        keras.layers.InputLayer(input_shape=input_shape),
-        keras.layers.Dense(output_shape),
-        keras.layers.Softmax()
+    layers = [
+        tf.keras.layers.InputLayer(input_shape=input_shape),
+        tf.keras.layers.Flatten(),
+    ]
+    if batch_norm:
+        layers.append(tf.keras.layers.BatchNormalization())
+    layers.extend([
+        tf.keras.layers.Dense(output_shape),
+        tf.keras.layers.Dense(output_shape),
```

Comment:
Assume the duplication was unintentional?

Reply:
Ah I see the same for `pytorch`, was an extra layer needed for extra tests?

Reply:
Yeah, the issue was there was only one layer that could be set to non-trainable and I felt we probably wanted some weights that were still trainable to make sure that the setting was doing what we want it to. I've replaced it in what I'm currently doing though because the model for checking for non-trainable layers is sufficiently complex now that it doesn't make sense to use the above fixture. i.e. I've added nested sequential layers etc...